utilities/backup/backup_engine.cc

```diff
@@ -1269,54 +1296,59 @@ IOStatus BackupEngineImpl::Initialize() {
+              work_item.progress_callback, &temp, work_item.dst_temperature,
+              &bytes_toward_next_callback, &result.size, &result.checksum_hex);
+
+          RecordTick(work_item.stats, BACKUP_READ_BYTES,
+                     IOSTATS(bytes_read) - prev_bytes_read);
+          RecordTick(work_item.stats, BACKUP_WRITE_BYTES,
+                     IOSTATS(bytes_written) - prev_bytes_written);
+
+          result.db_id = work_item.db_id;
+          result.db_session_id = work_item.db_session_id;
+          result.expected_src_temperature = work_item.src_temperature;
+          result.current_src_temperature = temp;
+          if (result.io_status.ok() && !work_item.src_checksum_hex.empty()) {
+            // unknown checksum function name implies no db table file checksum
+            // in db manifest; work_item.src_checksum_hex not empty means backup
+            // engine has calculated its crc32c checksum for the table file;
+            // therefore, we are able to compare the checksums.
+            if (work_item.src_checksum_func_name ==
+                    kUnknownFileChecksumFuncName ||
+                work_item.src_checksum_func_name == kDbFileChecksumFuncName) {
+              if (work_item.src_checksum_hex != result.checksum_hex) {
+                std::string checksum_info(
+                    "Expected checksum is " + work_item.src_checksum_hex +
+                    " while computed checksum is " + result.checksum_hex);
+                result.io_status = IOStatus::Corruption(
+                    "Checksum mismatch after copying to " + work_item.dst_path +
+                    ": " + checksum_info);
+              }
+            } else {
+              // FIXME(peterd): dead code?
+              std::string checksum_function_info(
+                  "Existing checksum function is " +
+                  work_item.src_checksum_func_name +
+                  " while provided checksum function is " +
+                  kBackupFileChecksumFuncName);
+              ROCKS_LOG_INFO(
+                  options_.info_log,
+                  "Unable to verify checksum after copying to %s: %s\n",
+                  work_item.dst_path.c_str(), checksum_function_info.c_str());
             }
-          } else {
-            // FIXME(peterd): dead code?
-            std::string checksum_function_info(
-                "Existing checksum function is " +
-                work_item.src_checksum_func_name +
-                " while provided checksum function is " +
-                kBackupFileChecksumFuncName);
-            ROCKS_LOG_INFO(
-                options_.info_log,
-                "Unable to verify checksum after copying to %s: %s\n",
-                work_item.dst_path.c_str(), checksum_function_info.c_str());
           }
+          work_item.result.set_value(std::move(result));
+        } else {
+          ROCKS_LOG_INFO(options_.info_log, "Unknown work item type: %" PRIu64,
```

Comment:
I think it would be more appropriate to populate a result with `io_status = Status::InvalidArgument(...)` so that we don't quietly, potentially allow an operation to succeed overall if part of it failed.